### PR TITLE
Added structure for noise classes

### PR DESCRIPTION
**Pull request by pandyah5**
This PR is meant to add adversarial noise to data. We are currently supporting:

- Normal distribution
- Gaussian distribution
- Laplacian distribution

### CODE REVIEW

**Review by ArashAhmadian — CHANGES_REQUESTED** [review]

File: src/nepiada/utils/noise.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
+from abc import ABC, abstractmethod
+
+class adversarial_noise_strategy(ABC):
```
[the remark this answers is not in the document]
Comment (pandyah5) [reply src/nepiada/utils/noise.py]: Good catch!

File: src/requirements.txt
Comment (Thanoshan) [inline src/requirements.txt:1]: Just checking, are all these requirements currently needed (I believe a bunch of them are dependencies of PettingZoo which makes sense).

**Review by Thanoshan — APPROVED** [review]
LGTM.

**Review by ArashAhmadian — APPROVED** [review]
LGTM

File: src/requirements.txt
Comment (pandyah5) [reply src/requirements.txt:1]: Just petting zoo things, I did a pip freeze on the venv

**Pull request merged**